libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs

```diff
@@ -179,21 +179,25 @@ public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext d
                 throw new ArgumentNullException(nameof(dc));
             }
 
-            // Recognize token
-            var recognized = await RecognizeTokenAsync(dc, cancellationToken).ConfigureAwait(false);
-
             // Check for timeout
             var state = dc.ActiveDialog.State;
             var expires = (DateTime)state[PersistedExpires];
             var isMessage = dc.Context.Activity.Type == ActivityTypes.Message;
-            var hasTimedOut = isMessage && DateTime.Compare(DateTime.Now, expires) > 0;
+            var isTimeoutActivityType = isMessage

```

Comment:
Consider adding a bit detailed comment here around what's going on and why this is here. I happen to remember now from the conversations yesterday, but months from now the details will be gone. Having a detailed comment around "Why is this here?" would be great.

Reply:
Added descriptive comment